packages/inline-edit/src/InlineEdit.tsx

```diff
@@ -0,0 +1,73 @@
+import React, {
+  FC, createRef, useState, useEffect, useCallback, memo,
+} from 'react';
+import { withTheme } from 'styled-components';
+
+
+import { getInlineEditUncontrolled } from './InlineEditUncontrolled';
+import { IInlineEditProps } from './interfaces';
+
+
+interface IInlineEditState {
+  isEditing?: boolean;
+}
+
+const getPureInlineEdit: <TFieldValue = string>() => FC<
+IInlineEditProps<TFieldValue> & IInlineEditState> = function f<TFieldValue>() {
+  return ({
+    startWithEditViewOpen = false,
+    onConfirm,
+    defaultValue,
+    readView,
+    editView,
+    ...rest
+  }: IInlineEditProps<TFieldValue>) => {
+    const editViewRef = createRef<HTMLElement>();
+    const [isEditing, setIsEditing] = useState(startWithEditViewOpen);
+
+    useEffect(() => {
+      if (startWithEditViewOpen && editViewRef.current) {
+        editViewRef.current.focus();
+      }
+    }, [startWithEditViewOpen, editViewRef]);
+
+    useEffect(() => {
+      if (isEditing && editViewRef.current) {
+        editViewRef.current.focus();
+      }
+    }, [isEditing, editViewRef]);
+
+    const handleConfirm = useCallback((value: string): void => {
+      setIsEditing(false);
+      onConfirm(value);
+    }, [onConfirm]);
+
+    const handleCancel = useCallback((): void => {
+      setIsEditing(false);
+      onConfirm('');
+    }, [onConfirm]);
+
+    const handleEditRequested = useCallback((): void => {
+      setIsEditing(true);
+    }, []);
+
+    const InlineEditUncontrolled = getInlineEditUncontrolled<TFieldValue>();
```

Comment:
I don't understand why you use `getInlineEditUncontrolled `and `getPureInlineEdit `instead of just using components.
`TFieldValue `complicates code understanding and looking unnecessary.

Reply:
Because we don't know what type of value will be used in editView prop while including this component in application (string, number, date, etc), so we need talk typescript about it throw generic <TFieldValue>